src/hooks/useRouteAssetId/useRouteAssetId.ts

```diff
@@ -11,13 +18,29 @@ export const useRouteAssetId = () => {
     const assetIdPathMatch = matchPath<{ chainId: string; assetSubId: string }>(location.pathname, {
       path: '/assets/:chainId/:assetSubId',
     })
-    if (!assetIdPathMatch?.params) return
 
-    const { chainId, assetSubId } = assetIdPathMatch.params
+    // Extract the foxAsset part of a /fox/:foxAsset route, can be 'fox', 'foxy' or undefined
+    const foxPageAssetIdPathMatch = matchPath<{ foxAsset?: string }>(location.pathname, {
+      path: '/fox/:foxAsset?',
+    })
+
+    if (foxPageAssetIdPathMatch) {
+      const foxAsset = foxPageAssetIdPathMatch?.params?.foxAsset ?? FoxPageDefaultAsset
+
+      if (FoxRoutePartToAssetId[foxAsset]) {
+        setAssetId(FoxRoutePartToAssetId[foxAsset])
+      }
+      return
+    }
+
+    if (assetIdPathMatch?.params) {
+      const { chainId, assetSubId } = assetIdPathMatch.params
 
-    // Reconstitutes the assetId from valid matched params
-    const assetId = `${chainId}/${assetSubId}`
-    setAssetId(assetId)
+      // Reconstitutes the assetId from valid matched params
+      const assetId = `${chainId}/${assetSubId}`
+      setAssetId(assetId)
+      return
+    }
```

Comment:
do this in a hook in the plugin, why is this stuff leaking into general app level hooks?

Reply:
With the current plugin architecture, it won't be possible to have this as its own hook (or if we did, we'd have to import it in `<AppContext />` either way, so we're back to square one plugin smell into general purpose app code situation).
Pushed 9550c464011fb1d0ace84078c962c52148846887 moving the matcher into plugins folder.